docs/proposals/Ashish_Mahajan_Versioning_Proposal.md

```diff
@@ -0,0 +1,241 @@
+
+3. **Historical Transparency**: Easy access to how DEI recommendations have evolved over time.
+
+
+### User Scenarios
+
+**Scenario 1: Conference Organizer - Version Comparison**
+Sunita is organizing a conference and needs to compare current and previous recommendations:
+
+- She visits discover-cookbook.numfocus.org and sees she's viewing v2.0
+- She notices the version selector in the navigation
+- She clicks to view v1.0 while viewing the "venue selection" section
+- The system navigates her to the equivalent section in v1.0
+- She can easily toggle between versions to compare approaches
+<br><br>
+![version-selector](togglebutton1.png)
+<br><br>
+
+**Scenario 2: First-time Visitor - Version Awareness**
+Karan discovers the cookbook through social media:
+
+- He lands on discover-cookbook.numfocus.org (latest version)
+- The version indicator shows he's viewing the current version (v2.0)
+- He's aware previous versions exist through the version selector
+- If he follows an old link to v1.0, he'll see a banner indicating newer content exists
+
+![Banner](banner.png)
+
+## Implementation
+
+My implementation follows a systematic approach aligned with industry standards for documentation versioning:
+
+### 1. Version Data Architecture
+
+I will implement a centralized version registry that defines all versions and their relationships. This architecture will:
+- Provide a single source of truth for version information
+- Support automatic version detection from URL paths
+- Enable contextual navigation between versions
+- Scale to accommodate future versions
+- Maintain independence between versions
+
+The version data structure will include:
+- Version metadata (name, display name, URL path, branch, release date, status)
+- Utility functions for version detection and navigation
+- Mapping data for equivalent content across versions
+
+### 2. Enhanced GitHub Actions Workflow
+
+I will configure an improved deployment workflow by enhancing the existing GitHub Actions setup to:
+- Automatically detect the source branch
+- Map branches to appropriate URL paths
+- Apply version-specific configurations
+- Deploy to the correct subfolder in gh-pages
+- Preserve the CNAME file for the custom domain
+
+The workflow will determine the target path based on the branch (main → /, master → /v1.0/) and deploy each version to its dedicated location while maintaining the shared domain.
+
+### 3. Version Selector UI Component
+
+I will implement an accessible, user-friendly version selector component:
+
+1. **Component Location**: Integrated into the Jupyter Book in the left sidebar
+2. **Implementation Strategy**:
+   - DOM-based insertion for compatibility with Jupyter Book
+   - Event-driven interaction handling
+   
+3. **UI Features**:
+   - Current version indicator
+   - Dropdown for all available versions
+   - Version-specific metadata (status)
+   - Visual distinction for latest version
+
+4. **Responsive Design**:
```

Comment:
seems like a stretch goal. I would rather not change the over arching design of the entire project.